### PR TITLE
Feature/app error screen doublespending

### PR DESCRIPTION
**Pull request by this-kramer**
## Reason for this PR:

- Add an screen when DSP prevented user token update
- Add error screen to prevent the app from the unlikely event of crashing and give the user a way of fixing this by deleting the basket
- Improve error messages sent by the incentive service
out-1

### CODE REVIEW

File: android/app/src/main/java/org/cryptimeleon/incentive/app/domain/usecase/ResetAppUseCase.kt [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import org.cryptimeleon.incentive.app.domain.IBasketRepository
+import org.cryptimeleon.incentive.app.domain.ICryptoRepository
+import org.cryptimeleon.incentive.app.domain.IPromotionRepository
+
+class ResetAppUseCase(
+    private val cryptoRepository: ICryptoRepository,
+    private val basketRepository: IBasketRepository,
+    private val promotionRepository: IPromotionRepository
+) {
+    suspend operator fun invoke() {
```
Comment (pschuermann97) [inline android/app/src/main/java/org/cryptimeleon/incentive/app/domain/usecase/ResetAppUseCase.kt:12]: does this also make the onboarding screen reappear?

File: android/app/src/main/java/org/cryptimeleon/incentive/app/ui/checkout/DsDetectedUi.kt [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        verticalArrangement = Arrangement.SpaceBetween,
+        horizontalAlignment = Alignment.CenterHorizontally
+    ) {
+        Column(
+            Modifier.weight(1f),
+            verticalArrangement = Arrangement.Center,
+            horizontalAlignment = Alignment.CenterHorizontally
+        ) {
+            Text(
+                "🧐",
```
Comment (pschuermann97) [inline android/app/src/main/java/org/cryptimeleon/incentive/app/ui/checkout/DsDetectedUi.kt:40]: this emoji seems quite underwhelming to me :)

File: services/incentive/src/main/java/org/cryptimeleon/incentive/services/incentive/error/BasketAlreadyPaidException.java [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package org.cryptimeleon.incentive.services.incentive.error;
+
+public class BasketAlreadyPaidException extends IncentiveServiceException {
+    public BasketAlreadyPaidException() {
+        super("");
```
Comment (pschuermann97) [inline services/incentive/src/main/java/org/cryptimeleon/incentive/services/incentive/error/BasketAlreadyPaidException.java:5]: doesnt this mean that no message will be displayed?

File: services/incentive/src/main/java/org/cryptimeleon/incentive/services/incentive/error/OnlineDoubleSpendingException.java [hunk 4]
```diff
@@ -0,0 +1,7 @@
+package org.cryptimeleon.incentive.services.incentive.error;
+
+public class OnlineDoubleSpendingException extends IncentiveServiceException {
+    public OnlineDoubleSpendingException() {
+        super("");
```
Comment (pschuermann97) [inline services/incentive/src/main/java/org/cryptimeleon/incentive/services/incentive/error/OnlineDoubleSpendingException.java:5]: same question here

**Review by pschuermann97 — APPROVED** [review]

File: android/app/src/main/java/org/cryptimeleon/incentive/app/domain/usecase/ResetAppUseCase.kt (hunk 1 above)
Comment (this-kramer) [reply android/app/src/main/java/org/cryptimeleon/incentive/app/domain/usecase/ResetAppUseCase.kt:12]: No, it starts with the loading screen again to load pp, keys, promotions, etc

File: android/app/src/main/java/org/cryptimeleon/incentive/app/ui/checkout/DsDetectedUi.kt (hunk 2 above)
Comment (this-kramer) [reply android/app/src/main/java/org/cryptimeleon/incentive/app/ui/checkout/DsDetectedUi.kt:40]: Do you have a recommendation for a more suitable one?

File: services/incentive/src/main/java/org/cryptimeleon/incentive/services/incentive/error/BasketAlreadyPaidException.java (hunk 3 above)
Comment (this-kramer) [reply services/incentive/src/main/java/org/cryptimeleon/incentive/services/incentive/error/BasketAlreadyPaidException.java:5]: This is service side, all errors from this class will automatically be translated into a 400 with a fixed error message

File: services/incentive/src/main/java/org/cryptimeleon/incentive/services/incentive/error/OnlineDoubleSpendingException.java (hunk 4 above)
Comment (this-kramer) [reply services/incentive/src/main/java/org/cryptimeleon/incentive/services/incentive/error/OnlineDoubleSpendingException.java:5]: See above :)
But other error code such that the app can tell what happened

**Pull request merged**